frontend/src/metabase/visualizations/echarts/cartesian/waterfall/option/axis.ts

```diff
@@ -71,7 +77,10 @@ function getYAxisFormatter(
   renderingContext: RenderingContext,
 ) {
   return (rowValue: RowValue) => {
-    const value = checkNumber(rowValue) - negativeTranslation;
+    let value = checkNumber(rowValue) - negativeTranslation;
+    if (settings["graph.y_axis.scale"] === "pow") {
+      value = value >= 0 ? Math.pow(value, 2) : -Math.pow(value, 2);
```

Comment:
We apply power transformation here, rather than relying on the `valueGetter` used by the orginal y-axis formatter, because we need to apply the `negativeTranslation` first.